### PR TITLE
Fix #204: Document new arrangement of run.logicalLocations property

### LINKED ISSUE #204: Document new arrangement of run.logicalLocations property
Using parentKey

### PR DESCRIPTION
**Pull request by ghost**
@michaelcfanning

### CODE REVIEW

File: document/CoreFormat.mdk [hunk 1]
```diff
@@ -474,43 +474,47 @@ the programming language in which the programmatic construct specified by that l
...
+~ IsoExample
+In this example, a result was detected in the C++ class `namespaceA::namespaceB::classC`. The `logicalLocations`
+object contains not only a property describing the class, but also properties describing its parents.
 ```
 "logicalLocations": {
-  "namespaceA::namespaceB::classC": [
-    {
-      "name": "namespaceA",
-      "kind": "namespace"
+    "namespaceA::namespaceB::classC": {
```
Comment (michaelcfanning) [inline document/CoreFormat.mdk:494]: we have to settle the module question. is this a valid fully qualified logical name? it is one that appears in a stack-frame. do we allow producers to make a call on the top-level item? that seems reasonable for two reasons: in some cases, the FQLN is reasonably expected to express the root module/binary, in other cases, it is not typical to do so. second, the logical location chain identifies all its components (meaning that a consumer can determine whether or not the module was included and go look for it elsewhere if that makes sense).

foo.dll!ns::type::blah()

File: document/CoreFormat.mdk [hunk 2]
```diff
@@ -2050,58 +2054,51 @@ which encodes the function signature in a manner that is compiler-dependent and
...
+### `kind` property { #logicalLocation-kind }
 
-A `logicalLocationComponent` object should contain a property named `kind`, whose value shall be one of the following strings,
+A `logicalLocation` object should contain a property named `kind`, whose value shall be one of the following strings,
 if any of those strings accurately describes the construct identified by this object:
 
-*   `"function"`
+*   `"function"` 
 *   `"member"`
 *   `"method"`
```
Comment (michaelcfanning) [inline document/CoreFormat.mdk]: 'method' was removed from the schema. it must argue its way back in. :)

File: document/CoreFormat.mdk [hunk 3]
```diff
@@ -2112,6 +2109,16 @@ if any of those strings accurately describes the construct identified by this ob
 
 If none of those strings accurately describes the construct, `kind` may contain any value specified by the analysis tool.
 
+### `parentKey` property { #logicalLocation-parentKey }
+
+If the logical location represented by the `logicalLocation` object is a nested logical location,
+then the the `logicalLocation` object shall contain a property named `parentKey` whose value is a string
```
Comment (michaelcfanning) [inline document/CoreFormat.mdk]: 'the the' repeated terms

File: document/CoreFormat.mdk (hunk 2 above)
Comment (ghost) [inline document/CoreFormat.mdk]: Fixed.

---

In reply to: [64121106](https://github.com/sarif-standard/sarif-spec/pull/213#discussion_r64121106) [](ancestors = 64121106)

File: document/CoreFormat.mdk (hunk 3 above)
Comment (ghost) [inline document/CoreFormat.mdk]: Fixed.

---

In reply to: [64121137](https://github.com/sarif-standard/sarif-spec/pull/213#discussion_r64121137) [](ancestors = 64121137)

File: document/CoreFormat.mdk [hunk 4]
```diff
@@ -2112,6 +2109,16 @@ if any of those strings accurately describes the construct identified by this ob
 
 If none of those strings accurately describes the construct, `kind` may contain any value specified by the analysis tool.
 
+### `parentKey` property { #logicalLocation-parentKey }
+
+If the logical location represented by the `logicalLocation` object is a nested logical location,
+then the the `logicalLocation` object shall contain a property named `parentKey` whose value is a string 
+that matches the property name of the parent logical locations's `logicalLocation` object within
+`run.logicalLocations` (&sect;[#run-logicalLocations]).
```
Comment (michaelcfanning) [inline document/CoreFormat.mdk:2116]: can we make this sentence clearer? 'the logical location's logicalLocation object' is particularly hard to parse. can't we simply say 'the string that matches the property name of the parent 'logicalLocation' object within 'run.logicalLocations'? maybe you have an even clearer alternate?

File: document/CoreFormat.mdk (hunk 1 above)
Comment (ghost) [inline document/CoreFormat.mdk:494]: I propose that we not require module to be at the root.

---

In reply to: [64120981](https://github.com/sarif-standard/sarif-spec/pull/213#discussion_r64120981) [](ancestors = 64120981)

**Comment by michaelcfanning** [comment]
+1 found a 'kind' that need deleting, made a suggestion for clarifying text.

File: document/CoreFormat.mdk (hunk 4 above)
Comment (ghost) [inline document/CoreFormat.mdk:2116]: Yours is good. Fixed.

---

In reply to: [64121288](https://github.com/sarif-standard/sarif-spec/pull/213#discussion_r64121288) [](ancestors = 64121288)
